modules/images/webp-uploads/fallback.js

```diff
@@ -4,21 +4,57 @@ window.wpPerfLab = window.wpPerfLab || {};
 	window.wpPerfLab.webpUploadsFallbackWebpImages = function( media ) {
 		for ( var i = 0; i < media.length; i++ ) {
 			try {
-				if ( ! media[i].media_details.sources || ! media[i].media_details.sources['image/jpeg'] ) {
-					continue;
-				}
-
-				var ext = media[i].media_details.sources['image/jpeg'].file.match( /\.\w+$/i );
-				if ( ! ext || ! ext[0] ) {
-					continue;
-				}
+				var image         = media[ i ],
+					media_details = image.media_details,
+					media_sources = media_details.sources,
+					sizes         = media_details.sizes,
+					sizes_keys    = Object.keys( sizes ),
+					images        = document.querySelectorAll( 'img.wp-image-' + image.id );
 
-				var images = document.querySelectorAll( 'img.wp-image-' + media[i].id );
 				for ( var j = 0; j < images.length; j++ ) {
-					images[j].src = images[j].src.replace( /\.webp$/i, ext[0] );
-					var srcset = images[j].getAttribute( 'srcset' );
+
+					var src    = images[ j ].src,
+						srcset = images[ j ].getAttribute( 'srcset' );
+
+					// If there are no sizes, then replace src through sources, there is nothing more to replace.
+					if ( media_sources ) {
+						src = src.replace( media_sources['image/webp'].file, media_sources['image/jpeg'].file );
+						images[ j ].setAttribute( 'src', src );
+						break;
+					}
```

Comment:
This if clause needs to be removed: You're still checking specifically the root `sources` value here, which based on your changes to the REST response shouldn't even be set. In addition, see my previous comments: We cannot assume that `src` is the full image, so this code here still looks off.

Reply:
The REST API returns a `sizes` array without any sizes, so I have to use `&& ! sizes.full` Then it works.